src/components/Articles/CompanyProvisions.vue

```diff
@@ -0,0 +1,227 @@
+<template>
+  <div id="company-provisions">
+    <confirm-dialog ref="confirmCompanyProvisionsDialog" attach="#company-provisions" />
+    <v-layout v-if="!isEditing">
+      <v-flex xs3>
+        <label><strong>Pre-existing Company Provisions</strong></label>
+        <v-chip v-if="hasProvisionsRemovedPropsChanged" x-small label color="#1669BB" text-color="white">
+          CHANGED
+        </v-chip>
+      </v-flex>
+      <v-flex xs7 class="info-text" v-if="provisionsRemoved">
+        The company has resolved that none of the Pre-existing Company Provisions are to apply to this company.
+      </v-flex>
+      <v-flex xs7 class="info-text" v-else>
+        This company has Pre-existing Company Provisions
+      </v-flex>
+      <v-flex mt-n2 xs2 class="align-right" v-if="!hasProvisionsRemovedPropsChanged">
+        <v-btn id="correct-company-provisions" text color="primary" @click="isEditing = true">
+          <v-icon small>mdi-pencil</v-icon>
+          <span>{{ editLabel }}</span>
+        </v-btn>
+      </v-flex>
+      <v-flex mt-n2 xs2 class="align-right" v-else>
+        <v-btn
+          id="undo-company-provisions"
+          text
+          color="primary"
+          class="undo-company-provisions"
+          @click="resetCompanyProvisions"
+        >
+          <v-icon small>mdi-undo</v-icon>
+          <span>Undo</span>
+        </v-btn>
+
+        <!-- More Actions Menu -->
+        <span class="more-actions">
+          <v-menu offset-y>
+            <template v-slot:activator="{ on }">
+              <v-btn text small color="primary" class="more-actions-btn" v-on="on">
+                <v-icon>mdi-menu-down</v-icon>
+              </v-btn>
+            </template>
+            <v-list>
+              <v-list-item class="actions-dropdown_item" @click="isEditing = true">
+                <v-list-item-subtitle>
+                  <v-icon small>mdi-pencil</v-icon>
+                  <span class="ml-1">{{ editLabel }}</span>
+                </v-list-item-subtitle>
+              </v-list-item>
+            </v-list>
+          </v-menu>
+        </span>
+      </v-flex>
+    </v-layout>
+    <v-layout v-else>
+      <v-flex xs3>
+        <label><strong>Pre-existing Company Provisions</strong></label>
+      </v-flex>
+      <v-flex xs9>
+        <v-layout>
+          <v-flex>
+            <p class="instructions-paragraph">
+              Complete this item only if the company has resolved that none of the Pre-existing Company Provisions
+              are to apply to this company (refer to Part 17 and Table 3 of the Regulation under the Business
+              Corporations Act).
+            </p>
+            <v-layout>
+                <v-checkbox
+                  id="cp-checkbox"
+                  class="mt-1"
+                  :color="isInvalid ? checkBoxColor : ''"
```

Comment:
Code smell (esp using a prop with a hard-coded hex color).

Could you make this component invalid/disabled instead, to use the built-in features?

Or, could you conditionally set a class (`:class=...`), which would specify the styles you want?

Reply:
I tried to set the style with a class but it didn't work. I checked the documentation for that component it says I should set the color prop.

Reply:
If you can't make this component invalid or disabled to achieve the style you want then OK, use the :color directive. But use the color name instead of a property to a hard-coded hex value.

Because this is a Vuetify component, it doesn't know about our app's [CSS variables](https://github.com/bcgov/business-edit-ui/blob/800fbf3cb3049cfbb739ca5e83570e1762e57313/src/assets/styles/theme.scss#L63). But it does know about the [Vuetify theme colors](https://github.com/bcgov/business-edit-ui/blob/800fbf3cb3049cfbb739ca5e83570e1762e57313/src/main.ts#L73) so use that name instead.

Reply:
Done. Thanks.